games/game_stalkeranomaly.py

```diff
@@ -132,20 +141,724 @@ def getContentsFor(self, tree: mobase.IFileTree) -> List[int]:
         return self.content
 
 
+class IVec3:
+    def __init__(self, x, y, z):
+        self.x = x
+        self.y = y
+        self.z = z
+
+    def __str__(self) -> str:
+        return f"{self.x}, {self.y}, {self.z}"
+
+    def set(self, x, y, z):
+        self.x = x
+        self.y = y
+        self.z = z
+
+
+class IVec4(IVec3):
+    def __init__(self, x, y, z, w):
+        super().__init__(x, y, z)
+        self.w = w
+
+    def __str__(self) -> str:
+        return f"{self.x}, {self.y}, {self.z}, f{self.w}"
+
+    def set(self, x, y, z, w):
+        super().set(x, y, z)
+        self.w = w
+
+
+class IFlag:
+    def __init__(self, flag: int):
+        self._flag = flag
+
+    def __str__(self) -> str:
+        return str(self._flag)
+
+    def assign(self, mask):
+        self._flag = mask
+
+    def has(self, mask) -> bool:
+        return bool((self._flag & mask) == mask)
+
+    def set(self, mask):
+        self._flag |= mask
+
+    def remove(self, mask):
+        self._flag &= ~mask
+
+
+class IReader:
+    def __init__(self, buffer: bytes):
+        self._buffer = buffer
+        self._pos = 0
+
+    def __len__(self) -> int:
+        return len(self._buffer)
+
+    def _read_(self, size: int):
+        pos = min(len(self._buffer), self._pos + size)
+        buffer = self._buffer[self._pos : pos]
+        return (buffer, pos)
+
+    def read(self, size: int = -1):
+        if size < 0:
+            size = len(self._buffer)
+        if len(self._buffer) <= self._pos:
+            return b""
+        (buffer, pos) = self._read_(size)
+        self._pos = pos
+        return bytes(buffer)
```

Comment:
Any reason to construct a new bytes object here? `bytes` are immutable in Python.

Reply:
No reason, just a mistake